src/samples/Teams/TeamsAgent/MyAgent.cs

```diff
@@ -0,0 +1,136 @@
+        {
+            RegisterExtension(new TeamsAgentExtension(this), tae =>
+            {
+                tae.OnMessageEdit(MessageEdited);
+                tae.MessageExtensions.OnSelectItem(OnSelectItem);
+                tae.MessageExtensions.OnQuery("findNuGetPackage", OnQuery);
+            });
+            OnMessageReactionsAdded(OnMessageReaction);
+            OnConversationUpdate(ConversationUpdateEvents.MembersAdded, WelcomeMessageAsync);
+            OnActivity(ActivityTypes.Message, OnMessageAsync);
+        }
+
+        private async Task<MessagingExtensionResult> OnSelectItem(ITurnContext turnContext, ITurnState turnState, object item, CancellationToken cancellationToken)
+        {
+            string jsonItem = JsonSerializer.Serialize(item);
+            await turnContext.SendActivityAsync("select item " + jsonItem, cancellationToken: cancellationToken);
+            return new MessagingExtensionResult() { ActivityPreview = new Activity() { Type = ActivityTypes.Message, Text = jsonItem } };
+        }
+
+        private async Task<MessagingExtensionResult> OnQuery(ITurnContext turnContext, ITurnState turnState, Query<IDictionary<string, object>> query, CancellationToken cancellationToken)
+        {
+
+            JsonElement el = query.Parameters.TryGetValue<JsonElement>("NuGetPackageName");
+
+            string text = el.GetString() ?? string.Empty;
+
+
+            IEnumerable<PackageItem> packages = await FindPackages(text);
+            List<MessagingExtensionAttachment> attachments = [.. packages.Select(package =>
+            {
+                string cardValue = $"{{\"packageId\": \"{package.Id}\", \"version\": \"{package.Version}\", \"description\": \"{PackageItem.NormalizeString(package.Description!)}\", \"projectUrl\": \"{package.ProjectUrl}\", \"iconUrl\": \"{package.IconUrl}\"}}";
+
+                ThumbnailCard previewCard = new() { Title = package.Id, Tap = new CardAction { Type = "invoke", Value = cardValue } };
+                if (!string.IsNullOrEmpty(package.IconUrl))
+                {
+                    previewCard.Images = [new CardImage(package.IconUrl, "Icon")];
+                }
+
+                MessagingExtensionAttachment attachment = new()
+                {
+                    ContentType = HeroCard.ContentType,
+                    Content = new HeroCard { Title = package.Id },
+                    Preview = previewCard.ToAttachment()
+                };
+
+                return attachment;
+            })];
+
+            return new MessagingExtensionResult
+            {
+                Type = "result",
+                AttachmentLayout = "list",
+                Attachments = attachments
+
+            };
+        }
+
+        private static async Task<IEnumerable<PackageItem>> FindPackages(string text)
+        {
+            JsonNode? obj = JsonObject.Parse(await new HttpClient().GetStringAsync($"https://azuresearch-usnc.nuget.org/query?q=id:{text}&prerelease=true"));

```

Comment:
I borrow this function from BF-SDK samples, and wondering if this should be updated to
1. use S.T.J instead of Json.net
2. Create the client from the available HttpClientFactory